bytefield.typ

```diff
@@ -189,15 +203,56 @@
   show_size: false,
 )
 
+#let annotation(pos, col:0, ..args, body) = (
+  type: "annotation",
+  pos: pos,
+  col: col,
+  args: args,
+  body: body
+)
+
```

Comment:
Nice, i had the same idea. Because then we can swap out tablex to the new buildin table implementation if it is ready without changing the bytefield api. 

Could you shortly describe what pos and col refers to?

Reply:
uh sorry, forced pushed a better example just this minute.
`pos` is the position left/right of the bytefield in the pre/post columns.
`col` specifies the exact column if you have multiple pre/post columns (see the new example)

Reply:
How about renaming these?
`pos` to `side`
`col` to simply `y` or `column`  ?

Reply:
`col` could also be a `level`
if the columns is calculated as distance from the bytefield. So it becomes more of hierarchical notion
`side:left, level:2` equals `pre.len() - level - 1`
And
`side:right, level:2` equals `pre.len() + bits + level`